modules/siddhi-core/src/test/java/org/wso2/siddhi/core/aggregation/Aggregation2TestCase.java

```diff
@@ -313,4 +313,130 @@ public void incrementalStreamProcessorTest50() throws InterruptedException {
+            InputHandler inputStreamInputHandler = siddhiAppRuntime.getInputHandler("inputStream");
+            siddhiAppRuntime.start();
+
+            //Wednesday, May 31, 2017 11:05:49 PM (add 5.30 to get corresponding IST time)
+            stockStreamInputHandler.send(new Object[]{"WSO2", 50f, 60f, 90L, 6});
+            // Thursday, June 1, 2017 4:05:50 AM (add 5.30 to get corresponding IST time)
+            stockStreamInputHandler.send(new Object[]{"WSO2", 50f, 60f, 90L, 6});
+
+            // Thursday, June 1, 2017 4:05:51 AM
+            stockStreamInputHandler.send(new Object[]{"IBM", 50f, 60f, 90L, 6});
+
+            // Thursday, June 1, 2017 4:05:52 AM
+            stockStreamInputHandler.send(new Object[]{"WSO2", 60f, 44f, 200L, 56});
+            stockStreamInputHandler.send(new Object[]{"WSO2", 100f, null, 200L, 16});
+
+            // Thursday, June 1, 2017 4:05:54 AM
+            stockStreamInputHandler.send(new Object[]{"IBM", 100f, null, 200L, 26});
+            stockStreamInputHandler.send(new Object[]{"IBM", 100f, null, 200L, 96});
+
+            // Thursday, June 1, 2017 4:05:56 AM
+            stockStreamInputHandler.send(new Object[]{"IBM", 900f, null, 200L, 60});
+            stockStreamInputHandler.send(new Object[]{"IBM", 500f, null, 200L, 7});
+
+            // Thursday, June 1, 2017 4:06:56 AM
+            stockStreamInputHandler.send(new Object[]{"IBM", 400f, null, 200L, 9});
+            stockStreamInputHandler.send(new Object[]{"IBM", 600f, null, 200L, 6});
+
+            // Thursday, June 1, 2017 4:07:56 AM
+            stockStreamInputHandler.send(new Object[]{"IBM", 600f, null, 200L, 6});
+
+            // Thursday, June 1, 2017 5:07:56 AM
+            stockStreamInputHandler.send(new Object[]{"IBM", 700f, null, 200L, 20});
+
+            Thread.sleep(100);
+            LocalDate currentDate = LocalDate.now();
+            String year = String.valueOf(currentDate.getYear() + 2);
+            inputStreamInputHandler.send(new Object[]{"IBM", 1, "2017-06-01 09:35:52 +05:30",
+                    year + "-06-01 09:35:52 +05:30", "hours"});
+            inputStreamInputHandler.send(new Object[]{"IBM", 1, "2017-06-01 09:35:52 +05:30",
+                    year + "-06-01 09:35:52 +05:30", "hours"});
+            Thread.sleep(100);
+
+            Event[] events1 =
+                    siddhiAppRuntime.query("from stockAggregation within 0L, " + (System.currentTimeMillis()
+                            + 1000000) + "L per 'hours' select AGG_TIMESTAMP, avgPrice, totalPrice as sumPrice");
+
+            Event[] events2 =
+                    siddhiAppRuntime.query("from stockAggregation within 0L, " + (System.currentTimeMillis()
+                            + 1000000) + "L per 'hours' select AGG_TIMESTAMP, avgPrice, totalPrice as sumPrice");
+
+            List<Object[]> expected = new ArrayList<>();
+            expected.add(inEventsList.get(1));
```

Comment:
For expected you have to give the values, in this case you are basically asserting the event produced with a clone of itself

Reply:
This is to check the two results after running the join query two times. In line 405 and 407 I am sending the same two events to the join query. Im verifying the first result data to the second result data.

Reply:
Then let's name them accordingly, firstJoinEvent, secondJoinEvent.. This can be even allocated before rather than copying to one variable and then processing